tuple-table/src/main/java/com/iluwatar/tupletable/MemberTupleDao.java

```diff
@@ -0,0 +1,149 @@
+package com.iluwatar.tupletable;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.sql.Types;
+
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class MemberTupleDao {
+  /**
+   * Method implemented to find the member based on the input param.
+   *
+   * @param memberNo is key with which tuple record is looked up
+   * @return member retrieved from the database
+   */
+  public MemberDto findMember(long memberNo) throws SQLException, ClassNotFoundException {
+    Database db = new Database();
+    MemberDto member = new MemberDto();
+    member.setMemberNumber(memberNo);
+
+    try (Connection con = db.getConnection();
+         PreparedStatement ps = con.prepareStatement("select fieldname, numerical, string "
+                 + "from object_data where obj_pk = ?")) {
+      // create a new table
+      ps.setLong(1, memberNo);
+      try (ResultSet rs = ps.executeQuery()) {
+        while (rs.next()) {
+          String fieldName = rs.getString(1);
+          String strVal = rs.getString(3);
+          if (strVal != null) {
+            setVal(fieldName, member, strVal);
+          } else {
+            //we do this indirectly to make database typecasting more reliable
+            long lngVal = rs.getLong(2);
+            if (!rs.wasNull()) {
+              setVal(fieldName, member, lngVal);
+            }
```

Comment:
Same here. Why do we need to set the value?

Reply:
Its setting the member DTO using the reflection technique in both places.